File: maven-simplest/src/main/java/ETL/src/Reducer.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

import org.json.JSONObject;

public class Reducer {

	private final static String DELIMITER = "@15619@delimiter@";
	private final static String NEWLINE = "@15619@newline@\n";

	public static void main(String[] args) {
		BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
		String line = null;
		try {		
			while((line = bufferedReader.readLine()) != null){
				try{
					//line = line.split("\t")[1];
					JSONObject jsonObj = new JSONObject(line);
					StringBuilder sb = new StringBuilder();
					sb.append(jsonObj.getString("uid"));
					sb.append(DELIMITER);
					sb.append(jsonObj.get("score"));
					sb.append(DELIMITER);
					sb.append(jsonObj.get("time"));
					sb.append(DELIMITER);
					sb.append(jsonObj.get("epoch"));
					sb.append(DELIMITER);
					sb.append(jsonObj.get("text"));
					sb.append(NEWLINE);
					System.out.print(sb.toString());
				} catch (Exception e){
					e.printStackTrace();
				}
			}
		} catch (IOException e) {
			e.printStackTrace();
		}	
	}
}

File: maven-simplest/src/main/python/ETL/reducer.py
#!/usr/bin/env python
from __future__ import print_function
import sys
import os
import happybase
import json

TABLE_NAME = "tweets"
DELIMITER = "<15619_delimiter>"
NEWLINE = "<15619_newline>"
#Remember to modify the DNS address of HBase master
DNS = "ec2-54-165-248-154.compute-1.amazonaws.com"
connection = happybase.Connection(DNS)

def create_table(table):
	global connection
	try:
		families = {'info': dict(in_memory=True, block_cache_enabled=True)}
		connection.create_table(table, families)
	except Exception, e:
		pass

def main():
	create_table(TABLE_NAME)
	table = connection.table(TABLE_NAME)
	with table.batch(batch_size=1000) as tweet_batch:
		for line in sys.stdin:
			try:
				tweet = json.loads(line)
				uid = tweet['uid'].encode('utf-8')
				tid = tweet['tid'].encode('utf-8')
				score = tweet['score'].encode('utf-8')
				time = tweet['time'].encode('utf-8')
				epoch = tweet['epoch'].encode('utf-8')
				text = tweet['text'].encode('utf-8')
				tweet_batch.put(
					tweet['uid'],
					{
						'info:score': score,
						'info:text': text,
						'info:tid': tid,
						'info:timestamp': time, 
						'info:epoch': epoch,
					})
				print(tid, uid, score, time, epoch, text, sep=DELIMITER, end=NEWLINE)
			except Exception, e:
				pass

main()